Added mc command for fetching details about a minecraft server

Iangecko opened the pull request:
Added mc command for getting minecraft server details.
![image](https://user-images.githubusercontent.com/38113821/61830908-60fc6280-ae3a-11e9-9324-5360f8e06ac9.png)

I think it would a fun command to add, so we can see if anyone is playing without having to boot minecraft and look at the server list.

Commit:
Added mc command for fetching details about a minecraft server

ntoskrnl4 commented on modules/mc.py:
```diff
@@ -0,0 +1,396 @@
+
+from client import client 
+from six import string_types
+import discord, socket, struct, io, base64, datetime, json, random
```
Per general readability guidelines among programming languages, it is much better to lay out every single import independently on their own lines, unless the imports are related. And just some nitpicks, please leave a newline between `from x import y` imports and `import x` imports, and alphabetize them.

ntoskrnl4 commented on modules/mc.py:
```diff
@@ -0,0 +1,396 @@
+
+        delta = (received - sent)
+        # We have no trivial way of getting a time delta :(
+        return (delta.days * 24 * 60 * 60 + delta.seconds) * 1000 + delta.microseconds / 1000.0
```
yes you do.

`delta.total_seconds()` returns a float with the number of seconds. It's [right in the documentation for the module](https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds)

ntoskrnl4 commented on modules/mc.py:
```diff
@@ -0,0 +1,396 @@
+    server = MinecraftServer.lookup(SERVER_URL) # Instantiate MinecraftServer class from SERVER_URL
+    status = server.status() # Get status, including icon, user count, usernames, etc
+    players = [] # Players on the server
+    description = "\n**Latency:** {}ms\n**Version:** {}\n**Players:** {}/{}".format(status.latency, status.raw["version"]["name"], status.raw["players"]["online"], status.raw["players"]["max"])
```
`.format()` is a mess, please use f-strings instead.


[How to use f-strings](https://www.youtube.com/watch?v=nghuHvKLhJA)

ntoskrnl4 commented on modules/mc.py:
```diff
@@ -0,0 +1,396 @@
+        embed.add_field(name="Players", value="\n".join(players))
+    
+    file = discord.File(fp=buffer, filename="icon.png")
+    await message.channel.send(file=file, embed=embed) # Send embed
```
Does this actually work for setting the thumbnail? The code reads like it will upload the embed, then the server icon separately, rather than adding the image into the embed. I don't know exactly how it works though, and I've never seen this before (it's not even in the d.py docs).

Also, per pep8, please add a trailing newline to your file.

Iangecko replied:
Yes, this the way you have to do it when uploading images for embeds, sadly. We had to do it with pyrobot and I don’t think it was improved in the rewrite. 

Added new line.

ntoskrnl4 commented on modules/mc.py:
```diff
@@ -0,0 +1,396 @@
+# mc.py a command for fetching details about a minecraft server
+
+from client import client 
+from six import string_types
```
Why are you using `string_types` from `six`? On my systems, this only returns `(str,)`, I'm not really sure what this is achieving. I'd recommend removing this import and just using the raw `str` class instead (the code looks like that's what it expects anyways).

ntoskrnl4 reviewed (changes requested):
I like the code, and I like the purpose it's achieving here. I'd like to see a couple of changes made though, primarily for readability. Message me on Discord if you want clarification or help with anything.

Commit:
Updated as per requested changes
reordered module imports
changed strings_types to standard str
updated time delta function
changed all embed format strings to f strings for readability
added new line at bottom in accordance with pep8

Iangecko commented:
Requested changes have been met, ready for final review and merge.

ntoskrnl4 commented:
Looks good. I'll merge and pull to the local repo.
